packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,302 @@
+  trigger={<Button content="A trigger" />}
+/>
+```
+
+## API proposal
+
+Dialog component is composed by 3 sub-components: Header, Body and Footer.
+
+### Components
+
+| Component    | Purpose                                                  |
+| ------------ | -------------------------------------------------------- |
+| Dialog       | The main wrapper component.                              |
+| DialogHeader | (optional) Component for the title and the close button. |
+| DialogBody   | (optional) Component for the main content of dialog.     |
+| DialogFooter | (optional) Component for the main actions of dialog.     |
+
+## Dialog
+
+The dialog is a container which handles styling (border, background etc.) and the overlay (dimmed background).
+
+### Anatomy
+
+![Visual anatomy of Dialog component](./assets/dialog-anatomy.png)
+
+### API
+
+| Property    | Values                        | Default   | Purpose                                    |
+| ----------- | ----------------------------- | --------- | ------------------------------------------ |
+| type        | `modal`, `non-modal`, `alert` | `modal`   | Dialog variations                          |
+| isOpen      | boolean                       | `false`   | Set to `true` when the dialog is visible   |
+| isDraggable | boolean                       | `false`   | Set to `true` to make the dialog draggable |
+| overlay     | _slot_                        | undefined | Dimmed background of dialog                |
+
+- `type` property (dialog variations):
+
+  - `modal`: When this type of dialog is open, the rest of the page is dimmed out and cannot be interacted with. The tab sequence is kept within the dialog and moving the focus outside the dialog will imply closing it. This is the default type of the component.
+
+  - `non-modal`: When a non-modal dialog is open, the rest of the page is not dimmed out and users can interact with the rest of the page. This also implies that the tab focus can move outside the dialog when it reaches the last focusable element.
+
+  - `alert`: are a special type of modal dialogs that interrupts the user's workflow to communicate an important message or ask for a decision. These dialogs are not dismissable, neither by escape key or by clicking outside the dialog.
+
+### DOM
+
+```html
+<div role="dialog" class="fui-dialog">{children}</div>
+```
+
+## Dialog Header
+
+The Header component will expect to have a dialog title/header and will show by default the close (X icon) button.
+
+### DOM
+
+```html
+<div class="fui-dialog-header">
+  <span>{title}</span>
+  <button aria-label="close" />
+</div>
+```
+
+### API
+
+| Property          | Values  | Default     | Purpose                         |
+| ----------------- | ------- | ----------- | ------------------------------- |
+| `closeButton`     | _slot_  | `undefined` | Button that closes the dialog   |
+| `hideCloseButton` | boolean | `false`     | Shows or hides the close button |
```

Comment:
**thought:** Wouldn't it make more sense to have only `closeButton` prop which is a slot. thus if user would like to hide this, they would use `closeButton={null}` or something like that. WDYT

Reply:
Well, `closeButton` is already `undefined` by default, right? So default behavior would be:

Produces no close button:
```
<Dialog>
  <DialogHeader>Header</DialogHeader>
</Dialog>
```

Hides close button due to effect:
```
<Dialog>
  <DialogHeader closeButton={canClose ? CloseButton : null}>Header</DialogHeader>
</Dialog>
```